src/reflect/scala/reflect/internal/pickling/UnPickler.scala

```diff
@@ -526,15 +526,6 @@ abstract class UnPickler {
       @inline def all[T](body: => T): List[T] = until(end, () => body)
       @inline def rep[T](body: => T): List[T] = times(readNat(), () => body)
 
-      // !!! What is this doing here?
-      def fixApply(tree: Apply, tpe: Type): Apply = {
-        val Apply(fun, args) = tree
-        if (fun.symbol.isOverloaded) {
-          fun setType fun.symbol.info
-          inferMethodAlternative(fun, args map (_.tpe), tpe)
```

Comment:
This was actually making things worse. Tree types are pickled, so `fun.tpe` is a non-overloaded type before this code. `inferMethodAlternative` is a no-op.

So in unpickled ASTs, a Select of an overloaded method had an overloaded `selectTree.tpe`. Trying to re-pickle such a tree fails (https://github.com/scala/bug/issues/13087).

Removing this coulde would be enough to fix the bug, but I added the new `fixOverload` anyway.